### DISCUSSION TITLE
Can I redirect console.log stuff to a separate log file in a plugin?

### DISCUSSION THREAD

**Discussion by mstachowsky** [atom/atom#22411] [Q&A]
I can use console.log and that works, but I'd like to save some data on my plugin's usage on my PC.  Is it possible to redirect console.log to a log file instead, or is there some other functionality I can use for that?

**Comment by icecream17** [comment]
Sure, just redefine `console.log` in your [init script](https://flight-manual.atom.io/hacking-atom/sections/the-init-file/)
```js
console.log = function () {
   // do something else
}
```

**Reply by mstachowsky** [reply]
Thanks! My big question, though, is how do I make it go where I want? Like, can I send it to a file?  How?

**Reply by icecream17** [reply]
Somehow atom supports `require`, so that means you can use [fs](https://nodejs.org/api/fs.html)

At the risk of helping too much (because google), here's some helper code
```js
const fs = require('fs')
const pathToYourFile = 'C:/absolute/path/to/your/file'

fs.appendFileSync(pathToYourFile, "text! more text")
```

<sup>I just noticed how `C:/` looks like an emoticon (in both directions!)</sup>

**Reply by ThatXliner** [reply]
> I just noticed how `C:/` looks like an emoticon (in both directions!)

You should see the `D` drive lmao
